mailscanner/user_manager.php

```diff
@@ -662,7 +708,13 @@ function change_state(id, filter) {
         END AS '" . safe_value(__('spamcheck12')) . "',
           spamscore AS '" . safe_value(__('spamscore12')) . "',
           highspamscore AS '" . safe_value(__('spamhscore12')) . "',
-        CONCAT('<a href=\"?token=" . $_SESSION['token'] . "&amp;action=edit&amp;key=',username,'\">" . safe_value(__('edit12')) . "</a>&nbsp;&nbsp;<a href=\"javascript:delete_user(\'',username,'\')\">" . safe_value(__('delete12')) . '</a>&nbsp;&nbsp;<a href="?token=' . $_SESSION['token'] . "&amp;action=filters&amp;id=',username,'\">" . safe_value(__('filters12')) . "</a>') AS '" . safe_value(__('action12')) . "'
+        CASE
+          WHEN login_expiry > " . time() . " THEN '" . safe_value(__('yes12')) . "'
+          WHEN login_expiry = 0 THEN '" . safe_value(__('unknown12')) . "'
+        ELSE 
+          '" . safe_value(__('no12')) . "'
+        END AS '" . safe_value(__('loggedin12')) . "',
+        CONCAT('<a href=\"?token=" . $_SESSION['token'] . "&amp;action=edit&amp;key=',username,'\">" . safe_value(__('edit12')) . "</a>&nbsp;&nbsp;<a href=\"javascript:delete_user(\'',username,'\')\">" . safe_value(__('delete12')) . '</a>&nbsp;&nbsp;<a href="?token=' . $_SESSION['token'] . "&amp;action=filters&amp;id=',username,'\">" . safe_value(__('filters12')) . "</a>&nbsp;&nbsp;<a href=\"javascript:logout_user(\'',username,'\')\">" . safe_value(__('logout12')) . "</a>') AS '" . safe_value(__('action12')) . "'
```

Comment:
This is one thing that is bugging me.  I might be being picky, though....

Logout shows as an option for all users, logged in or not.  I would like to just show logout for users actually logged in, but since the code that implements this is in a table built using a SQL query, I wasn't sure how to make the CONCAT selectively apply the Logout option based on whether a user is active.   Obviously, logging out a logged out user doesn't do anything and is completely harmless.